docs/development/new-packages.md

```diff
@@ -17,6 +17,18 @@ Most pure Python packages can be installed directly from PyPI with
 {func}`micropip.install` if they have a pure Python wheel. Check if this is the
 case by trying `micropip.install("package-name")`.
 
+Due to [current limitations of WebAssembly](https://pyodide.org/en/stable/usage/wasm-constraints.html),
+packages that make heavy use of multi-threading and multi-processing **may not** work
+without explicit support from maintainers. A workaround could be to avoid creating multiple
+threads when on WASM using a check like:
+
+```py
+is_wasm = sys.platform == "emscripten" or platform.machine() in ["wasm32", "wasm64"]
```

Comment:
It's always a bit annoying deciding how best to check this. `sys.platform == "emscripten"` implies `platform.machine() in ["wasm32", "wasm64"]` so it's a bit redundant. The Python test suite does:
```py
def _can_start_thread() -> bool:
    if sys.platform == "emscripten":
        return sys._emscripten_info.pthreads
    return platform.machine() not in ("wasm32", "wasm64")

can_start_thread = _can_start_thread()
```
it would be nice if threading/multiprocessing offered some API to query whether they work on the platform...

Reply:
I agree. I don't know what `sys._emscripten_info.pthreads` returns as if i understand threads are supported but due to spectre they were disabled. If I try to compile `argon2-cffi` with `-lpthread` it gives no issue, but the error persists.

I like your function more than mine, but maybe we should just remove the "suggested workaround" to not make any wrong assumptions. What do you think?